keras_cv/utils/preprocessing.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+
+def transform_value_range(images, original_range, target_range, dtype=tf.float32):
+    """transforms values in input tensor from original_range to target_range.
+
+    This function is intended to be used in preprocessing layers that
+    rely upon color values.  This allows us to assume internally that
+    the input tensor is always in the range [0, 255].
+
+    Args:
+        images: the set of images to transform to the target range range.
+        original_range: the value range to transform from.
```

Comment:
Do we need to have these ranges free?  Are we not going to practically support just 3 value ranges as in tests?

Reply:
We actually do need them to be free.  `value_range` will likely be a user supplied value and we can't make assumptions about their dataset/formatting.

Reply:
Ok I don't know what other ranges could be interesting other then `-1,1`, `0,1` and `0,255`. 
But If we want to have any free range is responsibility of the caller to validates the inputs as we could theoretically have `nan` in:

`images = (images - original_min_value) / (original_max_value - original_min_value)`